### PR TITLE
prov/hook/trace: Add trace log for CM operation APIs.

### PR DESCRIPTION
**Pull request by peinanz**
Add trace log on cm operations for ep/pep.
Remove postfix "_op" in trace cq ops, to be consistent with other trace ops.

Signed-off-by: Peinan Zhang <peinan.zhang@intel.com>

### CODE REVIEW

File: prov/hook/trace/src/hook_trace.c [hunk 1]
```diff
@@ -831,6 +837,101 @@ static struct fi_ops_tagged trace_tagged_ops = {
...
+	TRACE_CM(ret, myep->domain->fabric, (addr) ? (*(uint64_t *)addr) : 0);
+
+	return ret;
+}
+
+static int trace_listen(struct fid_pep *pep)
+{
+	struct hook_pep *mypep = container_of(pep, struct hook_pep, pep);
+
+    return fi_listen(mypep->hpep);
```
Comment (j-xiong) [inline prov/hook/trace/src/hook_trace.c:889]: Any reason these calls (listen/accept/reject/shutdown/join) are not traced? A single `TRACE_CM` might not fit all of them but you can always call `FI_TRACE` directly.

Comment (peinanz) [reply prov/hook/trace/src/hook_trace.c:889]: Is there any useful information we want to log for listen/accept/reject/shutdown/join ops?  These ops only have pointers to ep/pep.  Will such information be useful for debugging?

Comment (j-xiong) [reply prov/hook/trace/src/hook_trace.c:889]: Indeed, the details are carried in `param`, but those are provider specific so not much can be done other than printing the address and length, which may not be very useful. However, merely info of these functions being called can sometimes be useful.

File: prov/hook/trace/src/hook_trace.c [hunk 2]
```diff
@@ -831,6 +837,101 @@ static struct fi_ops_tagged trace_tagged_ops = {
 };
 
 
+static int trace_setname(fid_t fid, void *addr, size_t addrlen)
+{
+	int ret = 0;
+	struct hook_fabric *myfabric = hook_to_fabric(fid);
+
+	ret = fi_setname(hook_to_hfid(fid), addr, addrlen);
+	TRACE_CM(ret, myfabric, (addr) ? (*(uint64_t *)addr) : 0);
```
Comment (j-xiong) [inline prov/hook/trace/src/hook_trace.c:846]: What if addrlen > sizeof(uint64_t)?

Comment (peinanz) [reply prov/hook/trace/src/hook_trace.c:846]: Good catch, will fix the log of addr in all relevant cm ops. I thought the addr parameter is in the same type as in ep ops.

Comment (shefty) [reply prov/hook/trace/src/hook_trace.c:846]: Use ofi_straddr_log() to print the address in a readable format.  It only works with sockaddr, I think, but that may be the best we can do for now.

File: prov/hook/trace/src/hook_trace.c (hunk 1 above)
Comment (shefty) [reply prov/hook/trace/src/hook_trace.c:889]: indentation above is off.  I agree with @j-xiong , just knowing the calls were made can be useful.  Printing the ep pointers helps.

File: prov/hook/trace/src/hook_trace.c [hunk 3]
```diff
@@ -910,16 +1022,15 @@ static int trace_cq_signal_op(struct fid_cq *cq)
...
-	.readerr = trace_cq_readerr_op,
-	.sread = trace_cq_sread_op,
-	.sreadfrom = trace_cq_sreadfrom_op,
-	.signal = trace_cq_signal_op,
+	.read = trace_cq_read,
+	.readfrom = trace_cq_readfrom,
+	.readerr = trace_cq_readerr,
+	.sread = trace_cq_sread,
+	.sreadfrom = trace_cq_sreadfrom,
+	.signal = trace_cq_signal,
```
Comment (shefty) [inline prov/hook/trace/src/hook_trace.c:1030]: The CQ changes should be in a separate patch.

**Review by shefty — CHANGES_REQUESTED** [review]
